src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -790,13 +790,21 @@ void FancyZones::UpdateZoneWindows() noexcept
 void FancyZones::UpdateWindowsPositions() noexcept
 {
     auto callback = [](HWND window, LPARAM data) -> BOOL {
-        int i = static_cast<int>(reinterpret_cast<UINT_PTR>(::GetProp(window, ZONE_STAMP)));
-        if (i != 0)
+        size_t bitmask = reinterpret_cast<size_t>(::GetProp(window, MULTI_ZONE_STAMP));

```

Comment:
Not sure how often is UpdateWindowsPositions invoked, I just wonder if it would make sense to check for bitmask ! = 0 before executing the `for` loop to reduce the amount of processing for windows that are not zoned.
It's not something for this PR, but I also wonder if it would make sense to keep a list of the currently zoned windows instead of enumerating all the windows, not just for this call but also for other places or future functionalities.

Reply:
You're right, this for-loop should be inside the if block.